ocaml/gencert/lib.ml

```diff
@@ -23,14 +23,19 @@ let hostnames () =
   let hostname = Unix.gethostname () in
   let fqdns =
     Unix.getaddrinfo hostname "" [Unix.AI_CANONNAME]
-    |> List.filter_map (fun addrinfo ->
-           match addrinfo.Unix.ai_canonname with
-           | "" ->
-               None
-           | name ->
-               Some name)
+    |> List.map (fun x -> x.Unix.ai_canonname)
   in
-  match fqdns with [] -> [hostname] | fqdns -> fqdns
+  hostname :: fqdns
```

Comment:
I didn't include any lookups in /etc/hosts - I don't think this is necessary (`hostname :: fqnds` includes everything that I expected during testing). My justification is that self-signed certs of course work by default, when verification is disabled, and we expect users to create their own certificates anyway when enabling verification

Reply:
`Unix.getaddrinfo` resolves the hostnames to IP using `nsswitch.conf`, which uses currently `hosts:      files dns myhostname`, which means mappings in `/etc/hosts` takes precedence over dns resolution.

Reply:
The uniquify to this list was dropped on the `Selfcert.host` function, I think it should be reintroduced here to avoid duplicates..

Reply:
Indeed, so `getaddrinfo` is using /etc/hosts implicitly. I've reintroduced the uniquify here